### DISCUSSION TITLE
Cant lauch hyprland from display manager

### DISCUSSION THREAD

**Discussion by gibarel1** [hyprwm/Hyprland#1195] [Q&A] [answered]
I updated my laptop (arco linux/arch) and now i cant enter the desktop with hyprland from the display manager, i tried sddm, lightdm and emptty, all other DEs/WMs work fine, opening hyprland from a tty wroks as well, there are no logs when i enter with a session that isnt a hyprland (even if i attempt to login with it previously, there is no /tmp/hypr folder, enetering hyprland from terminal only shows the current sessions log). I dont know how i get sddm logs, but emptty gives me 2 errors (theres 2 hyprland sessions) one is "Error MIT-MACIG-1 key" and the other is about a "exec-hyprland" on .config/hypr/ not existing. i've unistalled hyprland and tried to install every option on the aur (stable, git and bin) all with the same issue.

Specs:
Distro: arco linux (arch)
CPU: i7-10510u
RAM: 8 gb
Kernel: 6.0.11 zen

**Comment by cwfryer** [comment]
Exact same problem here. Particularly challenging is the missing /tmp/hypr folder, which is missing even when using tty.
SDDM logs into journal by default.

Those logs show:
```
Dec 08 13:33:53 Surface3 sddm-greeter[1997]: Reading from "/usr/share/wayland-sessions/hyprland.desktop"
Dec 08 13:33:53 Surface3 sddm[1048]: Message received from greeter: Login
Dec 08 13:33:53 Surface3 sddm[1048]: Reading from "/usr/share/wayland-sessions/hyprland.desktop"
Dec 08 13:33:53 Surface3 sddm[1048]: Reading from "/usr/share/wayland-sessions/hyprland.desktop"
Dec 08 13:33:53 Surface3 sddm[1048]: Session "/usr/share/wayland-sessions/hyprland.desktop" selected, command: "~/.config/hypr/exec-hyprland"
Dec 08 13:33:53 Surface3 sddm-helper[2670]: [PAM] Starting...
Dec 08 13:33:53 Surface3 sddm-helper[2670]: [PAM] Authenticating...
Dec 08 13:33:53 Surface3 dbus-daemon[938]: [system] Activating via systemd: service name='org.freedesktop.home1' unit='dbus-org.freedesktop.home1.service' requested by ':1.30' (uid=0 pid=2670 comm="/usr/lib/sddm/sddm-helper --socket /tmp/sddm-auth6")
Dec 08 13:33:53 Surface3 sddm-helper[2670]: pam_systemd_home(sddm:auth): systemd-homed is not available: Unit dbus-org.freedesktop.home1.service not found.
Dec 08 13:33:53 Surface3 sddm-helper[2670]: [PAM] Preparing to converse...
Dec 08 13:33:53 Surface3 sddm-helper[2670]: [PAM] Conversation with 1 messages
Dec 08 13:33:53 Surface3 sddm-helper[2670]: gkr-pam: unable to locate daemon control file
Dec 08 13:33:53 Surface3 sddm-helper[2670]: gkr-pam: stashed password to try later in open session
Dec 08 13:33:53 Surface3 sddm-helper[2670]: pam_kwallet5(sddm:auth): pam_kwallet5: pam_sm_authenticate
Dec 08 13:33:53 Surface3 sddm-helper[2670]: [PAM] returning.
Dec 08 13:33:53 Surface3 sddm[1048]: Authenticated successfully
Dec 08 13:33:53 Surface3 sddm-greeter[1997]: Message received from daemon: LoginSucceeded
Dec 08 13:33:53 Surface3 sddm-helper[2670]: pam_kwallet5(sddm:setcred): pam_kwallet5: pam_sm_setcred
Dec 08 13:33:53 Surface3 sddm-helper[2670]: pam_unix(sddm:session): session opened for user (uid=1000) by (uid=0)
Dec 08 13:33:53 Surface3 systemd[1]: Created slice User Slice of UID 1000.
Dec 08 13:33:53 Surface3 sddm-helper[1540]: [PAM] Closing session
Dec 08 13:33:53 Surface3 sddm-helper[1540]: pam_unix(sddm-greeter:session): session closed for user sddm
Dec 08 13:33:53 Surface3 systemd[1]: Starting User Runtime Directory /run/user/1000...
**Dec 08 13:33:53 Surface3 systemd-logind[941]: New session 2 of user {me}.**
Dec 08 13:33:53 Surface3 sddm-helper[1540]: [PAM] Ended.
Dec 08 13:33:53 Surface3 systemd[1]: Finished User Runtime Directory /run/user/1000.
Dec 08 13:33:53 Surface3 sddm[1048]: Auth: sddm-helper exited successfully
Dec 08 13:33:53 Surface3 sddm[1048]: Greeter stopped.
Dec 08 13:33:53 Surface3 systemd[1]: Starting User Manager for UID 1000...
Dec 08 13:33:53 Surface3 systemd[1]: session-c1.scope: Deactivated successfully.
Dec 08 13:33:53 Surface3 systemd[1]: session-c1.scope: Consumed 1.729s CPU time.
Dec 08 13:33:53 Surface3 systemd-logind[941]: Session c1 logged out. Waiting for processes to exit.
Dec 08 13:33:53 Surface3 systemd-logind[941]: Removed session c1.
Dec 08 13:33:53 Surface3 dbus-daemon[938]: [system] Activating via systemd: service name='org.freedesktop.home1' unit='dbus-org.freedesktop.home1.service' requested by ':1.33' (uid=0 pid=2678 comm="(systemd)")
--
Dec 08 13:33:53 Surface3 systemd[2678]: Started GNOME Keyring daemon.
Dec 08 13:33:53 Surface3 gnome-keyring-daemon[2692]: GNOME_KEYRING_CONTROL=/run/user/1000/keyring
Dec 08 13:33:53 Surface3 systemd[2678]: Created slice User Core Session Slice.
Dec 08 13:33:53 Surface3 systemd[2678]: Starting D-Bus User Message Bus...
Dec 08 13:33:53 Surface3 systemd[2678]: Started D-Bus User Message Bus.
Dec 08 13:33:53 Surface3 sddm-helper[2670]: gkr-pam: unlocked login keyring
Dec 08 13:33:53 Surface3 sddm-helper[2670]: pam_kwallet5(sddm:session): pam_kwallet5: pam_sm_open_session
Dec 08 13:33:53 Surface3 sddm-helper[2715]: pam_kwallet5: final socket path: /run/user/1000/kwallet5.socket
**Dec 08 13:33:53 Surface3 sddm-helper[2670]: Starting: "/usr/share/sddm/scripts/wayland-session ~/.config/hypr/exec-hyprland"**
Dec 08 13:33:53 Surface3 sddm-helper[2718]: Jumping to VT 2
Dec 08 13:33:53 Surface3 sddm-helper[2718]: VT mode didn't need to be fixed
Dec 08 13:33:54 Surface3 sddm[1048]: Session started
Dec 08 13:33:54 Surface3 sddm-helper[2670]: [PAM] Closing session
Dec 08 13:33:54 Surface3 sddm-helper[2670]: pam_unix(sddm:session): session closed for user {me}
Dec 08 13:33:54 Surface3 sddm-helper[2670]: pam_kwallet5(sddm:session): pam_kwallet5: pam_sm_close_session
Dec 08 13:33:54 Surface3 sddm-helper[2670]: pam_kwallet5(sddm:setcred): pam_kwallet5: pam_sm_setcred
Dec 08 13:33:54 Surface3 sddm-helper[2670]: [PAM] Ended.
Dec 08 13:33:54 Surface3 sddm[1048]: Auth: sddm-helper exited with 127
Dec 08 13:33:54 Surface3 sddm[1048]: Socket server stopping...
Dec 08 13:33:54 Surface3 sddm[1048]: Socket server stopped.
Dec 08 13:33:54 Surface3 sddm[1048]: Display server stopping...
**Dec 08 13:33:54 Surface3 systemd-logind[941]: Session 2 logged out. Waiting for processes to exit.**
Dec 08 13:33:54 Surface3 sddm[1048]: Display server stopped.
Dec 08 13:33:54 Surface3 sddm[1048]: Running display stop script  "/usr/share/sddm/scripts/Xstop"
Dec 08 13:33:54 Surface3 sddm[1048]: Removing display ":0" ...
Dec 08 13:33:54 Surface3 sddm[1048]: Adding new display on vt 1 ...
Dec 08 13:33:54 Surface3 sddm[1048]: Loading theme configuration from ""
Dec 08 13:33:54 Surface3 sddm[1048]: Display server starting...
Dec 08 13:33:54 Surface3 sddm[1048]: Adding cookie to "/var/run/sddm/{0f15e97e-9cb5-40c5-993c-f50d288594e7}"
Dec 08 13:33:54 Surface3 sddm[1048]: Running: /usr/bin/X -nolisten tcp -background none -seat seat0 vt1 -auth /var/run/sddm/{0f15e97e-9cb5-40c5-993c-f50d288594e7} -noreset -displayfd 18
Dec 08 13:33:54 Surface3 acpid[933]: client 1108[0:0] has disconnected
Dec 08 13:33:54 Surface3 acpid[933]: client 1108[0:0] has disconnected
Dec 08 13:33:54 Surface3 acpid[933]: client connected from 2733[0:0]
Dec 08 13:33:54 Surface3 acpid[933]: 1 client rule loaded
Dec 08 13:33:54 Surface3 acpid[933]: client connected from 2733[0:0]
Dec 08 13:33:54 Surface3 acpid[933]: 1 client rule loaded
Dec 08 13:33:55 Surface3 sddm[1048]: Setting default cursor
Dec 08 13:33:55 Surface3 sddm[1048]: Running display setup script  "/usr/share/sddm/scripts/Xsetup"
```

So it looks like the same basic problem as you describe from emptty, it tries to exec ~/.config/hypr/exec-hyprland, fails, then logs out.

Specs:
Distro: arco linux (arch)
CPU: i3-4020Y
RAM: 4 gb
Kernel: 6.0.11 surface

**Comment by gibarel1** [comment]
`Dec 08 13:33:53 Surface3 sddm[1048]: Session "/usr/share/wayland-sessions/hyprland.desktop" selected, command: ~/.config/hypr/exec-hyprland`
maybe editing the entry and changing the command would solve it? dont know if it is the right approach though...
as i've said, i reinstalled hyprland many times to ry and solve this and also tried multiple display managers, and all other DEs i tried worked, so it's likely a problem with hyprland or the aur packages or arco linux. and it was working before the update so that makes it even weirder.

**Reply by vaxerski** [reply]
SDDM does not run as your user so using `~` most likely leads to a completely different directory than you would expect.

**Reply by gibarel1** [reply]
> SDDM does not run as your user so using `~` most likely leads to a completely different directory than you would expect.

any idea where it goes or how can we solve it?

**Reply by vaxerski** [reply]
use an absolute path and make sure every user is allowed to access and execute your wrapper.

**Reply by gibarel1** [reply]
i mean, there is no "exec-hyprland" on the ~/.config/hypr , so i'm not sure what to do here, it was working fine before updating and i'm using an aur package so everything should have been there, right?

**Reply by vaxerski** [reply]
there is no such thing provided by default by any AUR package nor Hyprland. It's either you or your distribution that set this up and I cannot provide support for that.

**Reply by gibarel1** [reply]
> there is no such thing provided by default by any AUR package nor Hyprland. It's either you or your distribution that set this up and I cannot provide support for that.

so as a last alternative, if i go to the desktop entry used by kde and change the command to `Hyprland`it should work, or do i need any arguments when launching Hyprland?

**Reply by vaxerski** [reply]
Please read the wiki:
https://wiki.hyprland.org/Getting-Started/Installation for installing and  https://wiki.hyprland.org/Getting-Started/Quick-start/ for a quick start.

it's there for a reason.

**Comment by gibarel1** [comment, marked as answer]
>Please read the wiki:
https://wiki.hyprland.org/Getting-Started/Installation for installing and https://wiki.hyprland.org/Getting-Started/Quick-start/ for a quick start.

thanks for pointing that to me. Here is how i fixed it: downloaded the latest release and looked for the .desktop entry under examples, comfirmed that only the `Hyprland`command was needed, went to /usr/share/wayland-sessions and edited Hyprland.desktop changed the "Exec=~/.config/hypr/exec-hyprland" to "Exec=Hyprland" and now it works. probably an issue with the package in the arcolinux repo, i first installed it from there and the entry was probably managed from there.

Thanks.
